mongo/integration/collection_test.go

```diff
@@ -74,6 +74,50 @@ func TestCollection(t *testing.T) {
 			assert.True(mt, ok, "expected error type %v, got %v", mongo.WriteException{}, err)
 			assert.NotNil(mt, we.WriteConcernError, "expected write concern error, got %+v", we)
 		})
+
+		mt.Run("options are converted", func(mt *mtest.T) {
+			// Require 3.2 servers for bypassDocumentValidation support.
+			nilOptsOpts := mtest.NewOptions().MinServerVersion("3.2")
+			nilOptsTestCases := []struct {
+				name            string
+				opts            []*options.InsertOneOptions
+				expectOptionSet bool
+			}{
+				{
+					"only nil is passed",
+					[]*options.InsertOneOptions{nil},
+					false,
+				},
+				{
+					"non-nil options is passed before nil",
+					[]*options.InsertOneOptions{options.InsertOne().SetBypassDocumentValidation(true), nil},
+					true,
+				},
+				{
+					"non-nil options is passed after nil",
+					[]*options.InsertOneOptions{nil, options.InsertOne().SetBypassDocumentValidation(true)},
+					true,
+				},
+			}
+
+			for _, testCase := range nilOptsTestCases {
+				mt.RunOpts(testCase.name, nilOptsOpts, func(mt *mtest.T) {
+					doc := bson.D{{"x", 1}}
+					_, err := mt.Coll.InsertOne(mtest.Background, doc, testCase.opts...)
+					assert.Nil(mt, err, "InsertOne error: %v", err)
+					optName := "bypassDocumentValidation"
+					evt := mt.GetStartedEvent()
+
+					val, err := evt.Command.LookupErr(optName)
+					if testCase.expectOptionSet {
+						assert.Nil(mt, err, "expected %v to be set but got: %v", optName, err)
+						assert.True(mt, val.Boolean(), "expected %v to be true but got: %v", optName, val.Boolean())
+					} else {
```

Comment:
nit: return early from the previous `if` statement instead of using `else`. There's no important reason for doing this AFAIK. It just aligns well with other tests we've written.

Reply:
Done.